pkg/awsutils/awsutils_test.go

```diff
@@ -380,11 +410,12 @@ func TestAllocENI(t *testing.T) {
 	mockEC2.EXPECT().AttachNetworkInterfaceWithContext(gomock.Any(), gomock.Any(), gomock.Any()).Return(attachResult, nil)
 	mockEC2.EXPECT().ModifyNetworkInterfaceAttributeWithContext(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil)
 
-	ins := &EC2InstanceMetadataCache{
+	cache := &EC2InstanceMetadataCache{
 		ec2SVC: mockEC2,
 		imds:   TypedIMDS{mockMetadata},
 	}
-	_, err := ins.AllocENI(false, nil, "")
+	cache.eventRecorder = setupEventRecorder(t)
```

Comment:
nit: can we set `eventRecorder` in constructor above? Seems easier to read that way. Same with the ones below